### PR TITLE
[libffi] Support arm/arm64

### PR DESCRIPTION
**Pull request by JackBoosY**
Fix arm/arm64 build, pass the correct arm / arm64 parameters to cmake.

Related: #10469.

### CODE REVIEW

**Comment by derselbst** [comment]
Could somebody pls. clarify why `CMAKE_SYSTEM_PROCESSOR` is used to determine the target? Whenever I try cross-compiling, it's set to `AMD64`, which is my host processor.

Also please note that `VCPKG_TARGET_ARCHITECTURE` could be used directly, so there should be no need to introduce a custom `ARCH_NAME` variable in the `CMakeLists.txt`.

File: ports/libffi/CMakeLists.txt [hunk 1]
```diff
@@ -21,13 +21,13 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "ARM")
     set(TARGET ARM)
 elseif(CMAKE_SYSTEM_NAME MATCHES "BSD" AND CMAKE_SIZEOF_VOID_P EQUAL 4)
     set(TARGET X86_FREEBSD)
-elseif(CMAKE_SYSTEM_NAME MATCHES "Windows" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "ARM")
+elseif(CMAKE_SYSTEM_NAME MATCHES "Windows" AND ARCH_NAME STREQUAL "arm")
```
Comment (derselbst) [inline ports/libffi/CMakeLists.txt]: Why do you use `ARCH_NAME`? You can just put

```cmake
elseif(CMAKE_SYSTEM_NAME MATCHES "Windows" AND VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
```

Comment (JackBoosY) [reply ports/libffi/CMakeLists.txt]: `VCPKG_TARGET_ARCHITECTURE` is not automatically passed into `CMakeLists.txt`, so we must manually pass this parameter.

Comment (derselbst) [reply ports/libffi/CMakeLists.txt]: According to the `config-arm-windows-out.log` it should be passed, using most recent vcpkg master:

https://ci.appveyor.com/project/derselbst/fluidsynth-g2ouw/builds/31479626/job/hm9u92nejl7083dx#L579

**Review by PhoebeHui — APPROVED** [review]

**Pull request merged**